src/Xamarin.iOS.ResourceDesigner/GenerateResourceDesignerTask.cs

```diff
@@ -37,6 +46,13 @@ public override bool Execute()
                 TrimmingPrefixes = ImageAssetsTrimmingPrefixes!.Split('|'),
                 FilenamesSeparatorChars = ImageAssetFilenamesSeparatorChars!.ToCharArray()
             };
+            
+            var colors = new ColorAssetRawDto
+            {
+                ColorAssetPaths = ColorAssets!.Select(asset => asset.GetMetadata("FullPath")).ToArray(),
+                TrimmingPrefixes = ColorAssetsTrimmingPrefixes!.Split('|'),
+                FilenamesSeparatorChars = ColorAssetFilenamesSeparatorChars!.ToCharArray()
+            };
```

Comment:
I see two possible options, with changes that I wrote before:
1. Use ImageAssetsRawDto here as well (since node is ImageAsset), remove ColorAssetRawDto at all
2. Rename it to ColorSetsRawDto and rename property to ColorSetsPaths (maybe even just Paths, not specify ColorSets, since the class name says it already). Also to make the similar renaming with ImageAssetsRawDto - rename it to ImageSetsRawDto and rename its Paths property respectively

Reply:
What do you think about this:
Create base record for nodes (InterfaceDefinition, Image asset nodes, Color asset nodes, etc) which contains Paths property
The InterfaceDefinition is inheritor

Then there is base assets record which contains prefixes and separators properties + get-only property which identifies the set extension (colorset, imageset, etc).
Then in execute we create colors and images vars based on data.